pkg/customizations/fsnode/fsnode.go

```diff
@@ -1,58 +1,68 @@
 package fsnode
 
 import (
+	"bytes"
+	"encoding/json"
 	"fmt"
 	"os"
-	"path"
+	"path/filepath"
 	"regexp"
+
+	"github.com/osbuild/images/internal/common"
 )
 
 const usernameRegex = `^[A-Za-z0-9_.][A-Za-z0-9_.-]{0,31}$`
 const groupnameRegex = `^[A-Za-z0-9_][A-Za-z0-9_-]{0,31}$`
 
+type baseFsNodeJSON struct {
```

Comment:
We could also call this `baseFsNodeData` might be a bit clearer

Reply:
This was exactly my idea when I read your concern about exposing fields.

Lesson learned, for the new blueprint I want to avoid getters/setters, so I think the new blueprint struct must be immutable. There might be some unmarshal hooks, but once everything is loaded up no changes should be allowed, no getters or anything. I really hope we can achieve this, then all fields can be exported. Well except unions which are not exported and one has to use `AsXXX` "getter" but that is the only way how to use it.